### PR TITLE
Phil DeFraties

### PR DESCRIPTION
**Pull request by PhilipDeFraties**
Phil DeFraties
I'm very irritated that this was so different and far more difficult than the practice assessment that I spent 3 hours reviewing.

### CODE REVIEW

File: lib/apartment.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class Apartment
+
+  attr_reader :number, :monthly_rent, :bathrooms, :bedrooms, :renter
+
+  def initialize(apartment_hash)
+    @number = apartment_hash[:number]
+    @monthly_rent = apartment_hash[:monthly_rent]
+    @bathrooms = apartment_hash[:bathrooms]
+    @bedrooms = apartment_hash[:bedrooms]
+    @renter = nil
```
Comment (mikedao) [inline lib/apartment.rb:10]: Good

File: lib/building.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  def add_unit(unit)
+    @units << unit
+  end
+
+  def renters
+    @units.map do |unit|
+      unit.renter.name
+    end
+  end
+
```
Comment (mikedao) [inline lib/building.rb:19]: chef's kiss

Comment (PhilipDeFraties) [reply lib/building.rb:19]: > chef's kiss

Mike i couldn't get this to work and I dont understand why. I pry'd between 15 and 16 and it said unit.renter.name was indeed the name I was looking for but it wouldn't return when running this block. I was stuck on this for 90 minutes.

Comment (mikedao) [reply lib/building.rb:19]: It should work - I would check your setup and how you are calling the method.

**Pull request closed without merging**